ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -77,18 +82,24 @@ type FcsErrorsStageProcessBase(fsFile, daemonProcess) =
         if isNotNull expr then highlightingCtor (expr, error.Message) :> _ else
         null
 
-    let createHighlighting (error: FSharpErrorInfo) (range: DocumentRange): IHighlighting =
+    let createHighlighting (error: FSharpErrorInfo) (range: DocumentRange): IHighlighting list =
         match error.ErrorNumber with
         | TypeEquation when error.Message.StartsWith(ifExprMissingElseBranch, StringComparison.Ordinal) ->
-            createHighlightingFromNodeWithMessage UnitTypeExpectedError range error 
+            [createHighlightingFromNodeWithMessage UnitTypeExpectedError range error]
```

Comment:
Can we somehow not add allocations for lists foe each error please? E.g think about passing the highlighting list or the collector instead, or somehow extracting cases where multiple errors are produced.

Btw, it's possible to have multiple highlightings for an error, please check redundant parens C# analyzer for reference.